client/state/connectionMiddleware/middleware.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+const connections = {};
```

Comment:
@lamosty when JavaScript added `const` it seemed non-intuitive how it applies to objects and arrays. However, because those variables are actually storing a reference to the array or object in memory, mutating the values in memory don't actually change the assignment to the named variable itself. Here is an example…

```js
const list = [1, 2, 3, 4];
var listCopy = list;

listCopy.push( 5 );

console.log( list ); // [1, 2, 3, 4, 5]
list === listCopy; // true
```

This is one of the major dangers in messing with a function's arguments because there's a false presumption of isolation.

```js
const favorites = { color: 'orange' };
const messWidIt = thing => {
	thing.icecream = 'vanilla';
	console.log( thing );
};

messWidIt( favorites ); // { color: 'orange', icecream: 'vanilla' }
console.log( favorites ); // { color: 'orange', icecream: 'vanilla' } 
```

So here we accidentally messed with the calling code from within the function. So we can modify/mutate arrays and objects that defined with `const` because we're not changing the reference. What _can't_ we do then?

```js
const list = [1, 2, 3, 4];
list = [1, 2, 3, 4, 5]; // nope
list = list.concat( 5 ); // nope - concat returns a new and cloned list
list = otherList; // nope
```

hope it helps!